src/main/java/org/owasp/benchmark/score/parsers/ShiftLeftReader.java

```diff
@@ -0,0 +1,67 @@
+package org.owasp.benchmark.score.parsers;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import org.owasp.benchmark.score.parsers.TestResults.ToolType;
+
+public class ShiftLeftReader extends Reader {
+  public TestResults parse(File f) throws IOException {
+
+    TestResults tr = new TestResults("ShiftLeft", true, ToolType.SAST);
+    BufferedReader reader = new BufferedReader(new FileReader(f));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *reader*. The resource is referenced in statements at the following lines: 13, 16. The resource closure statement is at line: 35. There are other execution paths that do not contain closure statements, e.g., when if-branch is taken at line 19. Either a) close *reader* in a try-finally block or b) close the resource by declaring *reader* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).